go.mod

```
@@ -3,70 +3,74 @@ module github.com/crunchydata/postgres-operator
 go 1.17
 
 require (
-	github.com/evanphx/json-patch v4.9.0+incompatible
+	github.com/evanphx/json-patch v5.6.0+incompatible
```

🌱 🤔 I see we use this package in one place to workaround an API bug. We should probably move off this `+incompatible` module path, but I want to think about what it takes to test it.

https://github.com/CrunchyData/postgres-operator/blob/dc8c01a6825ccb9150f48a553cb65bbea484273b/internal/controller/postgrescluster/apply.go#L98-L101

---

🌱 Our regular envtest tests cover this and pass. We can change this now if you'd like.

```suggestion
	github.com/evanphx/json-patch/v5 v5.6.0
```

And in `apply.go`,

```diff
--- internal/controller/postgrescluster/apply.go
+++ internal/controller/postgrescluster/apply.go
@@ -21,7 +21,7 @@ import (
        "fmt"
        "reflect"
 
-       jsonpatch "github.com/evanphx/json-patch"
+       jsonpatch "github.com/evanphx/json-patch/v5"
        "github.com/pkg/errors"
        appsv1 "k8s.io/api/apps/v1"
        batchv1 "k8s.io/api/batch/v1"
```